### DISCUSSION TITLE
Does this work with React 18?

### DISCUSSION THREAD

**Discussion by rostero1** [ZeeCoder/use-resize-observer#105] [Q&A]
React 18 will now batch all state updates, no matter where they come from.  Will this hook work with React 18 without things getting out of sync?

**Comment by ZeeCoder** [comment]
How React 18's concurrent mode works exactly, and whether it requires anything special of me as a lib author is not clear to me tbh.

I did make [some fixes](https://github.com/ZeeCoder/use-resize-observer/commit/852d976e481a215671be95964c6fa05825eee82a) a while back to make things work with v18, and haven't noticed issues since, but I can't really give any guarantees.

If you have any article recommendation to read up on the topic that'd be great, or even better: if anyone finds a related bug that's reproducible via a codesandbox that'd definitely help.
